config/opal_check_hwloc.m4

```diff
@@ -0,0 +1,160 @@
+               [AC_MSG_RESULT([yes])],
+               [AC_MSG_RESULT([no])
+                AC_MSG_ERROR([Cannot continue])])
+           AC_MSG_CHECKING([if hwloc version is lower than 2.0])
+           AS_IF([test "$opal_hwloc_dir" != ""],
+                 [opal_hwloc_CFLAGS_save=$CFLAGS
+                  CFLAGS="-I$opal_hwloc_dir/include $opal_hwloc_CFLAGS_save"])
+           AC_COMPILE_IFELSE(
+               [AC_LANG_PROGRAM([[#include <hwloc.h>]],
+                   [[
+#if HWLOC_API_VERSION >= 0x00020000
+#error "hwloc API version is greater or equal than 0x00020000"
+#endif
+                   ]])],
+               [AC_MSG_RESULT([yes])],
+               [AC_MSG_RESULT([no])
+                AC_MSG_ERROR([OMPI does not currently support hwloc v2 API
+Cannot continue])])
+           AS_IF([test "$opal_hwloc_dir" != ""],
+                 [CFLAGS=$opal_hwloc_CFLAGS_save])
+
+           # These flags need to get passed to the wrapper compilers
+
+           # Finally, add some flags to the wrapper compiler if we're
+           # building with developer headers so that our headers can
+           # be found.
+           hwloc_WRAPPER_EXTRA_CPPFLAGS=$opal_hwloc_CPPFLAGS
+           hwloc_WRAPPER_EXTRA_LDFLAGS=$opal_hwloc_LDFLAGS
+           hwloc_WRAPPER_EXTRA_LIBS=$opal_hwloc_LIBS
+
+           # Add some stuff to CPPFLAGS so that the rest of the source
+           # tree can be built
+           CPPFLAGS="$CPPFLAGS $opal_hwloc_CPPFLAGS"
+           LDFLAGS="$LDFLAGS $opal_hwloc_LDFLAGS"
+           LIBS="$LIBS $opal_hwloc_LIBS"
+           AS_IF([test "$OPAL_TOP_BUILDDIR" != "$OPAL_TOP_SRCDIR"],
+                 [CPPFLAGS="$CPPFLAGS -I$OPAL_TOP_BUILDDIR/opal/hwloc"])
+
+          # We have to do some extra indirection to get the
+          # OPAL_HWLOC_WANT_VERBS_HELPER to work.  First, the
+          # opal_hwloc_external_include file (set above), points to a
+          # file here in this component. That file will include the
+          # actual external hwloc.h file (via the
+          # MCA_hwloc_external_header define).  And if
+          # OPAL_HWLOC_WANT_VERBS_HELPER is set, that file will
+          # include the external hwloc/openfabrics-verbs.h file (via
+          # the MCA_hwloc_external_openfabrics_helper define).
+          AS_IF([test "$opal_hwloc_dir" != ""],
+                [opal_hwloc_include="$opal_hwloc_dir/include/hwloc.h"
+                 opal_hwloc_openfabrics_include="$opal_hwloc_dir/include/hwloc/openfabrics-verbs.h"],
+                [opal_hwloc_include="hwloc.h"
+                 opal_hwloc_openfabrics_include="hwloc/openfabrics-verbs.h"])
+          AC_DEFINE_UNQUOTED(hwloc_external_header,
+                 ["$opal_hwloc_include"],
+                 [Location of external hwloc header])
+          AC_DEFINE_UNQUOTED(hwloc_external_openfabrics_header,
+                 ["$opal_hwloc_openfabrics_include"],
+                 [Location of external hwloc OpenFabrics header])
+           $1],
```

Comment:
i do not think there are ```$1``` nor ```$2```any more.
since hwloc is mandatory, should we simply abort when it is not found ?

last but not least, travis config should be updated to install hwloc packages (both linux and osx) and ibm, lanl and mellanox should be requested to install hwloc packages.